### PR TITLE
Fix triangle count

### PR DESCRIPTION
**Pull request by ChuckHastings**
Fixed MG issue discovered by @jnke2016.

Triangle count test was not correct for MG.  If vertices are specified, the output size isn't known until after vertices are shuffled to the correct GPU.

### CODE REVIEW

**Comment by ChuckHastings** [comment]
rerun tests

File: cpp/tests/c_api/mg_triangle_count_test.c [hunk 1]
```diff
@@ -46,21 +46,28 @@ int generic_triangle_count_test(const cugraph_resource_handle_t* handle,
   cugraph_type_erased_device_array_t* p_start           = NULL;
   cugraph_type_erased_device_array_view_t* p_start_view = NULL;
 
+  int rank = 0;
+  rank = cugraph_resource_handle_get_rank(handle);
```
Comment (seunghwak) [inline cpp/tests/c_api/mg_triangle_count_test.c:50]: Sorry for nitpicking, but `int rank = cugraph_resource_handle_get_rank(handle);` might be cleaner.

**Review by seunghwak — APPROVED** [review]

File: cpp/tests/c_api/mg_triangle_count_test.c (hunk 1 above)
Comment (ChuckHastings) [reply cpp/tests/c_api/mg_triangle_count_test.c]: Fixed

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
rerun tests

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**